crates/cxx-qt-build/src/dependencies.rs

```diff
@@ -0,0 +1,301 @@
+) -> HashSet<PathBuf> {
+    dependencies
+        .iter()
+        .flat_map(|dep| dep.manifest.initializers.iter().cloned())
+        .chain(
+            interface
+                .iter()
+                .flat_map(|interface| interface.initializers.iter().cloned()),
+        )
+        .collect()
+}
+
+pub(crate) fn all_include_prefixes(
+    interface: &Interface,
+    dependencies: &[Dependency],
+) -> Vec<String> {
+    interface
+        .exported_include_prefixes
+        .iter()
+        .cloned()
+        .chain(
+            interface
+                .exported_include_directories
+                .iter()
+                .map(|(_path, prefix)| prefix.clone()),
+        )
+        .chain(
+            dependencies
+                .iter()
+                .flat_map(|dep| &dep.manifest.exported_include_prefixes)
+                .cloned(),
+        )
+        .collect()
+}
+
+pub(crate) fn reexported_dependencies(
+    interface: &Interface,
+    dependencies: &[Dependency],
+) -> Vec<Dependency> {
+    let mut exported_dependencies = Vec::new();
+    for link_name in &interface.reexport_links {
+        if let Some(dependency) = dependencies
+            .iter()
+            .find(|dep| &dep.manifest.link_name == link_name)
+        {
+            exported_dependencies.push(dependency.clone());
+        } else {
+            panic!("Could not find dependency with link name `{link_name}` to reexport!");
+        }
+    }
+    exported_dependencies
+}
+
+pub(crate) fn all_compile_definitions(
+    interface: Option<&Interface>,
+    dependencies: &[Dependency],
+) -> Vec<(String, Option<String>)> {
+    // For each definition, store the name of the crate that defines it so we can generate a
+    // nicer error message
+    let mut definitions: HashMap<String, (Option<String>, String)> = interface
+        .iter()
+        .flat_map(|interface| &interface.compile_definitions)
+        .map(|(key, value)| (key.clone(), (value.clone(), crate::crate_name())))
+        .collect();
+
+    for dependency in dependencies {
+        for (variable, value) in &dependency.manifest.defines {
+            use std::collections::hash_map::Entry::*;
+            let entry = definitions.entry(variable.to_owned());
+
+            match entry {
+                Vacant(entry) => {
+                    entry.insert((value.to_owned(), dependency.manifest.name.clone()));
+                }
+                Occupied(entry) => {
+                    let existing_value = &entry.get().0;
+                    if existing_value != value {
+                        panic!("Conflicting compiler definitions requested!\nCrate {existing} exports {variable}={existing_value:?}, and crate {conflicting} exports {variable}={value:?}",
+                            existing=entry.get().1,
+                            conflicting = dependency.manifest.name);
+                    }
+                    // else: Ignore Duplicate definition with the same value
```

Comment:
could have a reverse comment on top of the if instead

Reply:
Done